torchvision/datasets/forksafeziplookup.py

```diff
@@ -0,0 +1,46 @@
+import zipfile
+import threading
+import os, io
+import struct
+
+
+_FH_FILENAME_LENGTH = 10
+_FH_EXTRA_FIELD_LENGTH = 11
+STRUCT_FILE_HEADER = "<4s2B4HL2L2H"
+SIZE_FILE_HEADER = struct.calcsize(STRUCT_FILE_HEADER)
+
+
+# thread-safe/multiprocessing-safe
+class ForkSafeZipLookup:
```

Comment:
This needs tests, as it's pretty hard to have a sense if it works properly under multiple scenarios.
Could you add this to `datasets/utils.py`, and add corresponding tests to `test/test_dataset_utils.py`? You can create some random zip file in python, store it into a temporary file and assert for different cases.

Also, can you make it a new-style class for Python2, with `ForkSafeZipLookup(object)`?

Reply:
I'll work on the tests and moving it.

Reply:
This is done.